### ISSUE TITLE
An implementation of dynamic strings based on genvector

### ISSUE THREAD

**Issue by cher-nov** [cher-nov/genvector#3]
1. Should be null-terminated (i.e. `count` is always at least 1, and `length` is `count-1`).
2. Should have functions similar to standard ones from `string.h`, but with taking length into account.
3. Full name is "dynstring" and shortened is "dstr".
4. `typedef gvec_dstr_t dstr_t`
5. Use another repository for this.
6. Use `GVEC_MODULAR_APPROACH`.
7. Preserve compatibility with C89.
8. Alias gvec functions with `dstr_*` prefix using defines (nb: this will not break function pointer obtainment).

tl;dr: steal everything from [sds](https://github.com/antirez/sds)
